Lesson10_Task01.py

```diff
@@ -0,0 +1,29 @@
+"""
+Home-task Regex
+
+# open input.txt file and find all small
+ english letters that match such conditions:
+# target small letter round exactly with
+ three capital english letters on the left and on the right
+# Match examples:
+# sdTRYaUVKn  -> matches "a"
+# NTRSjARTb   -> no match ( not exactly 3 capital letters on the left)
+# zDFGbOPNq   -> matches "b"
+# Print Output as : "Result: "<your_found_human_readable_string">
+"""
+
+import re
+
+# Read the input file
+with open("input02.txt", "r") as input_file:
+    # Read the contents of the file
+    content = input_file.read()
+
+# Find all small English letters that match the specified conditions
+matches = re.findall(r'(?<![A-Z])[A-Z]{3}([a-z])[A-Z]{3}(?![A-Z])', content)
```

Comment:
If you have defined pre/post conditions better to write them fully. This will allow you do not use groups. Plus this adds readability. The best option compile regex with VERBOSE flag that allows you to add comments for each block.
Possible optimizations:
`r'(?<=[^A-Z][A-Z]{3})[a-z](?=[A-Z]{3}[^A-Z])'`